clients/unshaded/src/main/java/tachyon/client/file/options/SetStateOptions.java

```diff
@@ -62,15 +74,45 @@ public static SetStateOptions defaults() {
   }
 
   private final Boolean mPinned;
+  private final Long mTTL;
+
+  public SetStateOptions(SetStateTOptions options) {
+    mPinned = options.isSetPinned() ? options.isPinned() : null;
+    mTTL = options.isSetTtl() ? options.getTtl() : null;
+  }
 
   private SetStateOptions(SetStateOptions.Builder builder) {
     mPinned = builder.mPinned;
+    mTTL = builder.mTTL;
   }
 
   /**
    * @return the pinned flag value; it specifies whether the object should be kept in memory
    */
-  public Boolean getPinned() {
+  public Boolean getPinnedOrNull() {
     return mPinned;
   }
+
+  /**
+   * @return the TTL (time to live) value to use; it identifies duration (in milliseconds) the
+   *         created file should be kept around before it is automatically deleted, irrespective of
+   *         whether the file is pinned
+   */
+  public Long getTTLOrNull() {
+    return mTTL;
+  }
```

Comment:
I once used `Optional` from guava, but later jiri and I think `null` is fine. This class is the only exception from other option classes because it has to specify "when the option should have no effect".